Dockerfile

```diff
@@ -402,27 +402,25 @@ ARG CGO_CFLAGS="${CFLAGS}"
 ARG CGO_CXXFLAGS="${CXXFLAGS}"
 ARG CGO_LDFLAGS="${LDFLAGS}"
 
-WORKDIR /home/builder/go/src
+WORKDIR /home/builder/sdk-go/src
 RUN ./make.bash --no-clean
 
 # Build the standard library with and without PIE. Target binaries
 # should use PIE, but any host binaries generated during the build
 # might not.
-WORKDIR /home/builder/go
+WORKDIR /home/builder/sdk-go
+ENV GOOS="${GOOS}" \
+  GOROOT="/home/builder/sdk-go" \
+  GOPATH="/home/builder/gopath" \
+  PATH="/home/builder/sdk-go/bin:${PATH}"
```

Comment:
Maybe a quick comment to explain why these are set this way and here. Also, most importantly, why consuming stages probably _shouldn't_ need to set any of them.

```suggestion
# SDK Go toolchain environment; inherited by later build stages
ENV GOOS="${GOOS}"
ENV GOROOT="/home/builder/sdk-go"
ENV GOPATH="/home/builder/gopath"
# PATH is prefixed to select it over the host's Go tools.
ENV PATH="/home/builder/sdk-go/bin:${PATH}"
```